### PR TITLE
feat: add draopTarget feature

### PR DESCRIPTION
**Pull request by RostiMelk**
First of all, thank you for this great package! 
It makes it super simple to create a simple draggable interfaces without the need for adding a ton of boilerplate code. 

### PR Description

The idea is to add a placeholder, which would occupy the space of the available target position while dragging an item.
This would make it easier to see where the item would be placed, when dropping it.

This is a non-breaking change, as the placeholder is only added when the `placeholder` prop is set.

#### Preview
https://github.com/ValentinH/react-easy-sort/assets/25268506/59821aba-2b33-4da9-95f2-eedb3d99d903

### PR Checklist
- [x] Documentation added
- [x] Story added
- [x] Remove my own dev code (PNPM lockfile, package overrides, etc.)
_Any more?_

Would love to hear your thoughts on this, and the API for this feature!

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -2,12 +2,12 @@ import arrayMove from 'array-move'
...
 
 import { findItemIndexAtPosition } from './helpers'
-import { useDrag } from './hooks'
+import { useDrag, usePlaceholder } from './hooks'
 import { Point } from './types'
 
 const DEFAULT_CONTAINER_TAG = 'div'
 
-type Props<TTag extends keyof JSX.IntrinsicElements> = HTMLAttributes<TTag> & {
+type Props<TTag extends keyof JSX.IntrinsicElements> = Omit<HTMLAttributes<TTag>, 'placeholder'> & {
```
Comment (RostiMelk) [inline src/index.tsx:10]: We need to omit placeholder prop because it conflicts with the native HTMLAttribute

File: src/hooks.ts [hunk 2]
```diff
@@ -268,3 +268,72 @@ export const useDrag = ({
...
+      const newX = lockAxis === 'y' ? sourceRect.left : itemsRect[index].left
+      const newY = lockAxis === 'x' ? sourceRect.top : itemsRect[index].top
+
+      placeholderRef.current.style.transform = `translate3d(${newX}px, ${newY}px, 0px)`
+    }
+  }
+
+  const PlaceholderWrapper = (): React.ReactElement => {
+    return React.createElement(
+      'span',
```
Comment (RostiMelk) [inline src/hooks.ts:315]: Not sure if using span as a wrapper is ideal. The purpose of the wrapper is to automatically handle 'ref' additions, so end-users don't have to do it manually in their placeholder component

File: src/index.tsx [hunk 3]
```diff
@@ -157,6 +162,7 @@ const SortableList = <TTag extends keyof JSX.IntrinsicElements = typeof DEFAULT_
       }
 
       updateTargetPosition(pointInWindow)
+      placeholderLogic.show?.(sourceRect)
```
Comment (RostiMelk) [inline src/index.tsx:165]: All functions are optional, in case no placeholder prop is provided

File: src/hooks.ts [hunk 4]
```diff
@@ -268,3 +268,72 @@ export const useDrag = ({
...
+      {
+        ref: placeholderRef,
+        ariaHidden: true,
```
Comment (RostiMelk) [inline src/hooks.ts:318]: For avoiding screen readers from recognising this element

**Review by RostiMelk — COMMENTED** [review]
A few comments to clarify

**Comment by ValentinH** [comment]
Hi!
That looks like a nice feature, thanks!

Sorry for the delay, I don't have much time to allocate to this project these days, I'll try my best to review this ASAP.

File: src/index.tsx (hunk 1 above)
Comment (ValentinH) [reply src/index.tsx:10]: What do you think of calling this prop/feature: `dropTarget`?

File: src/index.tsx [hunk 5]
```diff
@@ -59,6 +62,8 @@ const SortableList = <TTag extends keyof JSX.IntrinsicElements = typeof DEFAULT_
   const lastTargetIndexRef = React.useRef<number | undefined>(undefined)
   // contains the offset point where the initial drag occurred to be used when dragging the item
   const offsetPointRef = React.useRef<Point>({ x: 0, y: 0 })
+  // contains the placeholder logic
+  const placeholderLogic = usePlaceholder(placeholder)
```
Comment (ValentinH) [inline src/index.tsx:66]: Nice way of encapsulating this new logic, well done!

File: src/hooks.ts [hunk 6]
```diff
@@ -268,3 +268,72 @@ export const useDrag = ({
...
+          left: 0,
+          pointerEvents: 'none',
+        },
+      },
+      content
+    )
+  }
+
+  return {
+    ref: placeholderRef,
```
Comment (ValentinH) [inline src/hooks.ts:333]: we don't use the ref outside of this hook so I don't think we need to return it.

File: src/hooks.ts [hunk 7]
```diff
@@ -268,3 +268,72 @@ export const useDrag = ({
...
+      const sourceRect = itemsRect[index]
+      const newX = lockAxis === 'y' ? sourceRect.left : itemsRect[index].left
+      const newY = lockAxis === 'x' ? sourceRect.top : itemsRect[index].top
+
+      placeholderRef.current.style.transform = `translate3d(${newX}px, ${newY}px, 0px)`
+    }
+  }
+
+  const PlaceholderWrapper = (): React.ReactElement => {
+    return React.createElement(
```
Comment (ValentinH) [inline src/hooks.ts:314]: let's rename this file to `hooks.tsx` and use JSX here :) 
BTW using a `<div>` looks more idiomatic to me for this kind of element.

**Review by ValentinH — COMMENTED** [review]
Only a few minor things, otherwise it looks good to me :)

**Comment by RostiMelk** [comment]
> Hi! That looks like a nice feature, thanks!
> 
> Sorry for the delay, I don't have much time to allocate to this project these days, I'll try my best to review this ASAP.

No worries! I just pushed a commit to the PR that solves all your comments :) 
Thanks for taking the time 🙏

File: src/index.tsx (hunk 1 above)
Comment (RostiMelk) [reply src/index.tsx]: Yeah that might make more sense. I updated all naming in code, including story and README as well.

**Review by ValentinH — APPROVED** [review]

**Comment by ValentinH** [comment]
Thank you, this is released ! 
https://github.com/ValentinH/react-easy-sort/releases/tag/v1.6.0

**Pull request merged**